manifests/init.pp

```diff
@@ -23,11 +23,13 @@
   String                        $user,
   String                        $filename             = ".netrc",
   Stdlib::Absolutepath          $file_path            = "$home_base_directory/$user/$filename",
-  Array                         $machine_user_password_triples) {
+  Array[Hash]                   $machine_user_password_triples) {
```

Comment:
Does this change allow the merge of keys like the IP or the token ?
I think as it is still an Array as main type, puppet will do a join of the lists.

Reply:
I'm thinking that it can be changed to this type:
`Hash[String, Hash]  $machine_user_password_triples`

and set in hiera like:

```
netrc:
  root_git_netrc:
    user:root
    home_base_directory: '/'
    machine_user_password_triples:
      git.code.tuenti.io:
        login: xxxx
        password: xxxxx
      github.com:
        login: yyyy
        password: yyyy
```
IMO it looks more representative in this way.
  
In that sense, when overwriting just a key in another role , it can be clearer.